src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -170,37 +176,68 @@ class NexusPublishPlugin : Plugin<Project> {
                         releaseTask {
                             mustRunAfter(publishAllTask)
                         }
-                        configureTaskDependencies(publishingProject, initializeTask, findStagingRepositoryTask, publishAllTask, closeTask, releaseTask, mavenRepo)
+                        configureTaskDependencies(publishingProject, initializeTask, findStagingRepositoryTask, publishAllTask, closeTask, releaseTask, publicationRepo, publicationType)
                     }
                 }
             }
             configureSimplifiedCloseAndReleaseTask(rootProject, extension)
         }
     }
 
-    private fun addMavenRepositories(project: Project, extension: NexusPublishExtension, registry: Provider<StagingRepositoryDescriptorRegistry>): Map<NexusRepository, MavenArtifactRepository> {
-        return extension.repositories.associateWith { nexusRepo ->
-            project.the<PublishingExtension>().repositories.maven {
-                name = nexusRepo.name
-                setUrl(
-                    project.provider {
-                        getRepoUrl(nexusRepo, extension, registry)
-                    }
-                )
-                val allowInsecureProtocol = nexusRepo.allowInsecureProtocol.orNull
-                if (allowInsecureProtocol != null) {
-                    if (GradleVersion.current() >= GradleVersion.version("6.0")) {
-                        isAllowInsecureProtocol = allowInsecureProtocol
-                    } else {
-                        project.logger.warn("Configuration of allowInsecureProtocol=$allowInsecureProtocol will be ignored because it requires Gradle 6.0 or later")
-                    }
-                }
-                credentials {
-                    username = nexusRepo.username.orNull
-                    password = nexusRepo.password.orNull
-                }
+    private fun addPublicationRepositories(
+        project: Project,
+        extension: NexusPublishExtension,
+        registry: Provider<StagingRepositoryDescriptorRegistry>
+    ): Map<NexusRepository, ArtifactRepository> = extension.repositories.associateWith { nexusRepo ->
+        createArtifactRepository(extension.publicationType.get(), project, nexusRepo, extension, registry)
+    }
+
+    private fun createArtifactRepository(
+        publicationType: PublicationType?,
+        project: Project,
+        nexusRepo: NexusRepository,
+        extension: NexusPublishExtension,
+        registry: Provider<StagingRepositoryDescriptorRegistry>
+    ): ArtifactRepository = when (publicationType!!) {
```

Comment:
That's a good point , I need to check if Property<T> has some guard rails that I can use

Reply:
Using convention would solve this issue, but that would require bumping minimum version to gradle 6, lmk if that is ok (also @marcphilipp ), otherwise I just add a nullcheck somewhere

Reply:
Gradle 6+ compatibility is completely fine.

Reply:
Technically builds can also override the convention to be `null` so we should ensure a good error message is created if it is `null` in any case. I don't think this alone justifies dropping compatibility with Gradle 5 although I'd be open to that in general.

Reply:
Marc is rigth explicit null can still be a problem, I will fix that quickly